netty/src/test/java/io/grpc/transport/netty/NettyClientHandlerTest.java

```diff
@@ -176,14 +198,16 @@ public void cancelShouldSucceed() throws Exception {
   }
 
   @Test
-  public void cancelBeforeStreamAssignedShouldSucceed() throws Exception {
+  public void cancelWhileBufferedShouldSucceed() throws Exception {
     handler.connection().local().maxActiveStreams(0);
     handler.write(ctx, new CreateStreamCommand(grpcHeaders, stream), promise);
     mockContext();
-    verify(stream, never()).id(any(Integer.class));
-    when(stream.id()).thenReturn(null);
-
-    handler.write(ctx, new CancelStreamCommand(stream), promise);
+    ArgumentCaptor<Integer> idCaptor = ArgumentCaptor.forClass(Integer.class);
+    verify(stream).id(idCaptor.capture());
+    when(stream.id()).thenReturn(idCaptor.getValue());
+    ChannelPromise cancelPromise = mock(ChannelPromise.class);
+    handler.write(ctx, new CancelStreamCommand(stream), cancelPromise);
+    verify(cancelPromise).setSuccess();
```

Comment:
@buchgr I guess that's fine, although it does seem a little strange.  @ejona86 any thoughts on this?